### PR TITLE
fix(caldav): decode values before returning

### LINKED ISSUE #28162: Calendar: Invitations aren't sent if Nextcloud user has "+" sign in email address
If the Nextcloud user has a "+" sign in his email address (e.g. "hans+nextcloud@example.com") configured within Nextcloud, calendar invitations to other attendees (no matter if external or internal users) aren't sent, even though
- The mail server is configured correctly (i.e. sending test mails is working),
- "Sending event invitations" is activated in groupware settings and
- The calendar isn't a shared calendar
(I list these three topics since they were mentioned in a lot of posts I read when I was looking for a solution to this problem). There's no hint in the UI that emails weren't sent and I couldn't find any related/relevant log entries in the server logfiles. 

### Steps to reproduce
1. Change email address of Nextcloud user to something with "+" in it
2. Create calendar event with this user's personal calendar and invite other users/email addresses

### Server configuration

**Operating system:** Ubuntu 20.04.2 LTS (GNU/Linux 5.4.0-80-generic x86_64)

**Web server:** nginx/1.18.0 (Ubuntu)

**Database:** Ver 15.1 Distrib 10.3.29-MariaDB

**PHP version:** 7.4.3

**Nextcloud version:** 21.0.3.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated (consecutively from Nextcloud 18)

**Where did you install Nextcloud from:** Nextcloud.com

**Signing status:** "No errors have been found."

**List of activated apps:**

Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - calendar: 2.3.1
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contacts: 4.0.0
  - contactsinteraction: 1.2.0
  - dav: 1.17.1
  - documentserver_community: 0.1.11
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_markdown: 2.3.4
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_texteditor: 2.14.0
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - forms: 2.2.4
  - groupfolders: 9.0.2
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - mail: 1.10.1
  - nextcloud_announcements: 1.10.0
  - notes: 4.1.0
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - onlyoffice: 7.0.2
  - password_policy: 1.11.0
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - tasks: 0.14.0
  - text: 3.2.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - admin_audit
  - dashboard
  - encryption
  - files_external
  - recommendations
  - user_ldap

**Nextcloud configuration:**
"system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "bla.blubb.de",
            "192.168.178.100"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "share_folder": "\/Mit mir geteilt",
        "dbtype": "mysql",
        "version": "21.0.3.1",
        "overwrite.cli.url": "http:\/\/bla.blubb.de",
        "overwriteprotocol": "https",
        "overwritehost": "bla.blubb.de",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "logtimezone": "Europe\/Berlin",
        "skeletondirectory": "",
        "maintenance": false,
        "mysql.utf8mb4": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "loglevel": 0,
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "DE",
        "remember_login_cookie_lifetime": 43200,
        "session_lifetime": 900,
        "auto_logout": true,
        "theme": "",
        "mail_smtpdebug": true
    }

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Any

**Operating system:** Any

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
* Resolves: #28162

Proper iTip message is now generated

Screenshot 2024-07-17 172429

### CODE REVIEW

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -130,6 +130,11 @@ protected function getAddressesForPrincipal($principal) {
 		if ($result === null) {
 			$result = [];
 		}
+		
+		// iterate through items and html decode values
+		foreach ($result as $key => $value) {
+			$result[$key] = urldecode($value);
```
Comment (miaulalala) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php:136]: To limit unknown side effects, please limit the urldecode to `sender` and `recipient`

**Review by miaulalala — CHANGES_REQUESTED** [review]

**Review by st3iny — APPROVED** [review]
Tested and works. I reproduced the initial issue before applying the PR and it is fixed now.

**Comment by st3iny** [comment]
For reference, this is how the array looks before and after decoding:

| Before | After |
| --- | --- |
| Screenshot_20240718_132742 | Screenshot_20240718_132752 |

**Review by miaulalala — APPROVED** [review]

**Pull request merged**